evo_researcher/benchmark/benchmark.py

```diff
@@ -43,6 +45,12 @@ def __init__(
         else:
             self.predictions = PredictionsCache(predictions={})
 
+        self.only_cached = only_cached
+        self.markets: list[Market] = [
+            m for m in markets 
+            if all(self.predictions.has_market(agent_name=agent.agent_name, question=m.question) for agent in self.registered_agents)
+        ] if self.only_cached else markets
```

Comment:
Add an option to filter down markets only to those available in all agents' caches. It's good for just re-running the benchmark after some small code changes.

Reply:
Not sure I understand this - if you've made some changes to an agent, you don't want use its cached results. I would have thought the opposite!

Reply:
If you made changes to the agent, then yes, but if you do changes to the benchmark itself (new metric, bug fix, etc.), using cached-only predictions is a way faster.

Reply:
Ah, understood, makes sense.